### PR TITLE
PRO-4775 Intestacy definition updates

### PR DESCRIPTION
**Pull request by Wivenhoe-HMCTS**
PRO-4775 Intestacy definition updates
```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: app/components/submit-data.js [hunk 1]
```diff
@@ -42,6 +45,17 @@ const dataMap = {
     caseId: 'ccdCase.id',
     caseState: 'ccdCase.state',
     registry: 'registry',
+    caseType: 'caseType',
+    deceasedHasAssetsOutsideUK: 'assets.assetsoverseas',
+    foreignAsset: 'assets.assetsoverseas',
```
Comment (LucaDelBuonoHMCTS) [inline app/components/submit-data.js]: I'll pretend I haven't seen this 🙈

File: app/components/submit-data.js [hunk 2]
```diff
@@ -76,6 +90,15 @@ const submitData = (ctx, data) => {
     mappedData.executorsApplying = executorsWrapper.executorsApplying(true);
     mappedData.executorsNotApplying = executorsWrapper.executorsNotApplying(true);
 
+    if (get(data, 'deceased.maritalStatus')) {
+        mappedData.deceasedMartialStatus = MartialStatusEnum.getCCDCode(get(data, 'deceased.maritalStatus'));
```
Comment (LucaDelBuonoHMCTS) [inline app/components/submit-data.js]: martial or marital? 😁

File: app/utils/MaritalStatusEnum.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const content = require('app/resources/en/translation/deceased/maritalstatus');
+
+class MartialStatusEnum {
```
Comment (LucaDelBuonoHMCTS) [inline app/utils/MaritalStatusEnum.js]: Martial law? 😄

File: app/utils/RelationshipToTheDeceasedEnum.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        case content.optionSpousePartner:
+            return this.getPartner();
+        case content.optionChild:
+            return this.getChild();
+        case content.optionAdoptedChild:
+            return this.getAdoptedChild();
+        case content.optionOther:
+            return this.getOther();
+        default:
+            throw new Error(`Enumerator RelationshipToDeceasedEnum value: ${value} not found`);
```
Comment (LucaDelBuonoHMCTS) [inline app/utils/RelationshipToTheDeceasedEnum.js]: Intent one level the cases please

File: app/utils/SpouseNotApplyingEnum.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    static getCCDCode(value) {
+        switch (value) {
+        case content.optionRenouncing:
+            return this.getRenouncing();
+        case content.optionOther:
+            return this.getOther();
+        default:
+            throw new Error(`Enumerator SpouseNotApplyingEnum value: ${value} not found`);
```
Comment (LucaDelBuonoHMCTS) [inline app/utils/SpouseNotApplyingEnum.js]: Indent

File: app/utils/MaritalStatusEnum.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        case content.optionMarried:
+            return this.getMarriedCivilPartnerhip();
+        case content.optionDivorced:
+            return this.getDivorcedCivilPartnerShip();
+        case content.optionSeparated:
+            return this.getJudicially();
+        case content.optionNotMarried:
+            return this.getNeverMarried();
+        default:
+            throw new Error(`Enumerator MartialStatusEnum value: ${value} not found`);
```
Comment (LucaDelBuonoHMCTS) [inline app/utils/MaritalStatusEnum.js]: Indent

File: test/unit/testMaritalStatusEnum.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+const MartialStatusEnum = require('app/utils/MaritalStatusEnum');
+const content = require('app/resources/en/translation/deceased/maritalstatus');
+const expect = require('chai').expect;
+
+describe('MaritalStatusEnum.js', () => {
+
+    describe('getMartialStatusCCDCode()', () => {
+        it('should return Widowed value', (done) => {
+            expect(MartialStatusEnum.getCCDCode(content.optionWidowed)).to.equal(MartialStatusEnum.getWidowed());
+            done();
```
Comment (LucaDelBuonoHMCTS) [inline test/unit/testMaritalStatusEnum.js]: Curfew as well?

File: app/components/submit-data.js (hunk 1 above)
Comment (Wivenhoe-HMCTS) [reply app/components/submit-data.js]: These are not the assets your looking for!! So says Femi anyway..

File: app/components/submit-data.js (hunk 2 above)
Comment (Wivenhoe-HMCTS) [reply app/components/submit-data.js]: changed

File: app/utils/MaritalStatusEnum.js (hunk 3 above)
Comment (Wivenhoe-HMCTS) [reply app/utils/MaritalStatusEnum.js]: changed

File: app/utils/MaritalStatusEnum.js (hunk 6 above)
Comment (Wivenhoe-HMCTS) [reply app/utils/MaritalStatusEnum.js]: changed

File: app/utils/RelationshipToTheDeceasedEnum.js (hunk 4 above)
Comment (Wivenhoe-HMCTS) [reply app/utils/RelationshipToTheDeceasedEnum.js]: changed

File: app/utils/SpouseNotApplyingEnum.js (hunk 5 above)
Comment (Wivenhoe-HMCTS) [reply app/utils/SpouseNotApplyingEnum.js]: changed

File: test/unit/testMaritalStatusEnum.js (hunk 7 above)
Comment (Wivenhoe-HMCTS) [reply test/unit/testMaritalStatusEnum.js]: changed

**Comment by Wivenhoe-HMCTS** [comment]
Jenkins build keeps failing

**Pull request closed without merging**